app/controllers/int_api/marketplaces_controller.rb

```diff
@@ -51,7 +51,7 @@ def create
 
     #include port separately to url_params if it was present
     if marketplace[:url] && marketplace[:url].split(':')[2]
-      url_params.merge!(port: marketplace[:url].split(':')[2])
+      url_params[:port] = marketplace[:url].split(':')[2]
```

Comment:
@ovan thanks for tip! I also felt bit overly complicated at that point, but somehow didn't think that some library could give the port nicely even thought the URL was just a text variable at that point.

Somehow I notice still this tendency to like passing stuff as objects that can nicely tell about their contents, and when all is passed as simple data types I feel like my tool box is much emptier. But I should learn better the tools that do nice things with simple datatypes, as that seems to be the direction were heading.. :)